src/analyticsdf/__init__.py

```diff
@@ -0,0 +1,77 @@
+"""Top-level package for analyticsdataframe."""
+
+import numpy as np
+import pandas as pd
+import contextlib
+import random
+
+@contextlib.contextmanager
+def set_random_state(random_state):
+    """Context manager for managing the random state.
+
+    Args:
+        random_state (int or np.random.RandomState):
+            The random seed or RandomState.
+        set_model_random_state (function):
+            Function to set the random state on the model.
+
+    """
+    original_state = np.random.get_state()
+
+    np.random.set_state(random_state.get_state())
+
+    try:
+        yield
+    finally:
+        current_random_state = np.random.RandomState()
+        current_random_state.set_state(np.random.get_state())
+        np.random.set_state(original_state)
```

Comment:
1. Yes! `current_random_state` is useless here, I'll get rid of it. Thanks!
2. I use `try/finally` here, so that the random state gets restored even if there's an exception raised in the with block that's using this context manager.
@EliWangC Thanks so much for the detailed review!